### ISSUE TITLE
 TestCase not comply with psr-4 autoloading standard

### ISSUE THREAD

**Issue by krishat2017** [BKWLD/laravel-pug#90]
I tried to install laravel-pug with following command
  composer require bkwld/laravel-pug
During installation got following message. Please let me know how to fix this.
```
Class Phug\Util\TestCaseTypeBase located in ./vendor/phug/phug/src/Phug/Util/Util/TestCaseTyped.php does not comply with psr-4 autoloading standard. Skipping.
Class Phug\Util\TestCaseTypeBase located in ./vendor/phug/phug/src/Phug/Util/Util/TestCaseUntyped.php does not comply with psr-4 autoloading standard. Skipping.

> php artisan vendor:publish --provider="Bkwld\LaravelPug\ServiceProvider"
Unable to locate publishable resources.
```
If it helps, I am running on 
  Ubuntu 18.04
  PHP 8.0.5
  Composer 2.0.8
In my composer.json I noticed following line in 'require ', block : bkwld/laravel-pug": "^2.3"

Thanks

**Comment by kylekatarnls** [comment]
Hello 👋 

Thanks for the report, `TestCaseTypeBase` is a new test util we needed for PHP 8 compatibility on our test suite but it's not needed user side and should not impact the install according to composer rules.

Can you try to run `composer require phug/phug:1.8.1` and retry to run `composer require bkwld/laravel-pug`?

**Comment by krishat2017** [comment]
Hi
Thanks a lot for your support. I run both the commands as you suggested and it installed without any error message.

**Comment by kylekatarnls** [comment]
Hi again, I just try to install `bkwld/laravel-pug` in a fresh install of Laravel 8, and it does not trigger any error message, only the skip messages you mention:

```
Class Phug\Util\TestCaseTypeBase located in ./vendor/phug/phug/src/Phug/Util/Util/TestCaseTyped.php does not comply with psr-4 autoloading standard. Skipping.
Class Phug\Util\TestCaseTypeBase located in ./vendor/phug/phug/src/Phug/Util/Util/TestCaseUntyped.php does not comply with psr-4 autoloading standard. Skipping.
```

Those are not error messages, just simple notice telling you those files were not loaded and as it's test utile, there is nothing to worry about.

But the install works fine, and Laravel load pug view without any issues so I recommend you remove `"phug/phug": "1.8.1"` from your **composer.json** and run `composer update`

In the next version I will move those files outside the autoloaded folder so it won't display any message but be aware this will not be an issue for you app if those file are skipped.

For the record, blocking errors occurring on composer appear in big red rectangle and the program exit with a non-`0` status code.

**Comment by kylekatarnls** [comment]
It's now fixed with phug/phug 1.8.3

**Comment by krishat2017** [comment]
Thanks for the explanation and the fix.
